astropy/__init__.py

```diff
@@ -113,96 +112,9 @@ class Conf(_config.ConfigNamespace):
-    constructed and passed to `pytest.main`.
-
-    Parameters
-    ----------
-    package : str, optional
-        The name of a specific package to test, e.g. 'io.fits' or 'utils'.
-        If nothing is specified all default Astropy tests are run.
-
-    test_path : str, optional
-        Specify location to test by path. May be a single file or
-        directory. Must be specified absolutely or relative to the
-        calling directory.
-
-    args : str, optional
-        Additional arguments to be passed to `pytest.main` in the `args`
-        keyword argument.
-
-    plugins : list, optional
-        Plugins to be passed to `pytest.main` in the `plugins` keyword
-        argument.
-
-    verbose : bool, optional
-        Convenience option to turn on verbose output from py.test. Passing
-        True is the same as specifying `-v` in `args`.
-
-    pastebin : {'failed','all',None}, optional
-        Convenience option for turning on py.test pastebin output. Set to
-        'failed' to upload info for failed tests, or 'all' to upload info
-        for all tests.
-
-    remote_data : bool, optional
-        Controls whether to run tests marked with @remote_data. These
-        tests use online data and are not run by default. Set to True to
-        run these tests.
-
-    pep8 : bool, optional
-        Turn on PEP8 checking via the pytest-pep8 plugin and disable normal
-        tests. Same as specifying `--pep8 -k pep8` in `args`.
-
-    pdb : bool, optional
-        Turn on PDB post-mortem analysis for failing tests. Same as
-        specifying `--pdb` in `args`.
-
-    open_files : bool, optional
-        Fail when any tests leave files open.  Off by default, because
-        this adds extra run time to the test suite.  Requires the
-        ``psutil`` package.
-
-    parallel : int, optional
-        When provided, run the tests in parallel on the specified
-        number of CPUs.  If parallel is negative, it will use the all
-        the cores on the machine.  Requires the `pytest-xdist` plugin.
-
-    docs_path : str, optional
-        The path to the documentation .rst files.
-
-    skip_docs : bool, optional
-        When `True`, skips running the doctests in the .rst files.
-
-    repeat : int, optional
-        If set, specifies how many times each test should be run. This is
-        useful for diagnosing sporadic failures.
-
-    See Also
-    --------
-    pytest.main : py.test function wrapped by `run_tests`.
-
-    """
-    test_runner = _get_test_runner()
-    return test_runner.run_tests(
-        package=package, test_path=test_path, args=args,
-        plugins=plugins, verbose=verbose, pastebin=pastebin,
-        remote_data=remote_data, pep8=pep8, pdb=pdb,
-        open_files=open_files, parallel=parallel, docs_path=docs_path,
-        skip_docs=skip_docs, repeat=repeat)
+# Create the test() function
+from .tests.runner import TestRunner
+test = TestRunner.make_test_runner_in(__path__[0])
```

Comment:
For the affiliated package template we may need something *slightly* ugly here along the lines of:

```python
try:
    from .tests.runner import TestRunner
    test = TestRunner.make_test_runner_in(__path__[0])
except ImportError:
    def test(*args, **kwargs):
        """Run the package test suit.  Requires the astropy package to be installed."""
        raise RuntimeError("The astropy package must be installed in order to run the tests.")
```